crates/ide_assists/src/handlers/generate_function.rs

```diff
@@ -73,6 +73,10 @@ fn gen_fn(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
                 get_fn_target(ctx, &target_module, call.clone())?
             }
             Some(hir::PathResolution::Def(hir::ModuleDef::Adt(adt))) => {
+                if let hir::Adt::Enum(_) = adt {
+                    return None;
+                }
+
```

Comment:
We shouldn't disable this assist for enums in general, we should disable it if the type is an enum and the function name is `PascalCase`, that is the function name starts with an uppercase letter

Reply:
Likewise the `generate_enum_variant` assist should only trigger on enums when the function name starts with an uppercase letter